lib/database.js

```diff
@@ -90,7 +90,8 @@ var database = {
 	getNextBatch : function( afterQueryFunction ) {
 		var query = 'SELECT `blog_id`, `monitor_url`, `site_status`, `last_status_change` ' +
 			'FROM `jetpack_monitor_sites` WHERE `bucket_no` >= ' +
-			fromBucketNo + ' AND `bucket_no` < ' + toBucketNo + ' AND `monitor_active` = 1;';
+			fromBucketNo + ' AND `bucket_no` < ' + toBucketNo + ' AND `monitor_active` = 1 AND ' +
+			'MOD(MINUTE(NOW()) + `jetpack_monitor_site_id`, `check_interval`) = 0;';
```

Comment:
I haven't tested that, but a spidey sense is telling me that there's a chance to skip items if the minute rolls over before we manage to fetch all buckets. For example, what would happen if we start the loop over buckets at 00:00:58 and it takes more than two seconds to move over the next bucket? 

Wouldn't it skip over items that are supposed to run at the `00:00` minute in the second (and further) buckets? Or am I not parsing this correctly.

Reply:
We ended up discussing this outside of Github. I'll add the key discussion points here so that there is a record.

This logic definitely has a risk of skipping items if a specific round takes longer than one minute to complete. In this scenario, the sites to be checked in that next minute will be skipped. With this approach, monitoring becomes essential as we need to ensure that max round times do not get too close to one minute.

To add some assurances that sites with high priority (ones with check intervals of one minute) are not skipped if the system starts to slow down, @bisko is working on a change that will interrupt a slow round when the next minute begins in order to prepend existing check queues with sites to be checked on one-minute intervals. This is still a theoretical solution and has not been tested. If this approach fails, we will develop another solution to ensure that a laggy host doesn't not cause skipped check intervals for priority sites.